### DISCUSSION TITLE
Multi class metrics

### DISCUSSION THREAD

**Discussion by nishchaychawla** [mlflow/mlflow#6646] [Q&A] [answered]
What is the most appropriate way to log and visualise field-level metrics in multi-class classification across different runs. For example in case of NER, if we want to see precision, recall, f1 for every entity

**Comment by vincentkoc** [comment, marked as answer]
@nishchaychawla https://mlflow-extend.readthedocs.io/en/latest/reference/plotting.html#mlflow_extend.plotting.confusion_matrix

**Reply by JHibbard** [reply]
I'd suggest combining koconder's answer with nested runs. You can have the top-level run produce a confusion matrix using information from each nested run. Hope that helps! There's an example of creating a nested run under the API docs for `mlflow.start_run`, you just have to scroll down a little from [[here](https://mlflow.org/docs/latest/python_api/mlflow.html#mlflow.start_run)].
